x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/DownsampleShardIndexer.java

```diff
@@ -148,6 +148,17 @@ class DownsampleShardIndexer {
 
     public DownsampleIndexerAction.ShardDownsampleResponse execute() throws IOException {
         final Query initialStateQuery = createQuery();
+        if (this.dimensions == null || this.dimensions.length <= 0) {
```

Comment:
I expect this to not be really necessary because creating a tsdb index without dimensions is not possible. and that would be the only way I see to have empty dimensions.
Anyway I think adding this check is ok udner the assumption that eventually failing those tasks is not an issue and does not create other issues.

Reply:
Tsdb indices created before 8.13.0 can get the dimensions for the the tsid? Since on earlier version the tsid isn't hashed and it contains the dimension field values pairs.

Reply:
Yes, I brought back this old logic in https://github.com/elastic/elasticsearch/pull/106878/commits/132e5f12e0a926d706e89230b8485a6e08236234